modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -37,9 +37,9 @@ void
 ComputeReturnMappingStress::initialSetup()
 {
   std::vector<MaterialName> models = getParam<std::vector<MaterialName> >("return_mapping_models");
-  for (unsigned i = 0; i < models.size(); ++i)
+  for (unsigned int i = 0; i < models.size(); ++i)
   {
-    MooseSharedPointer<RecomputeRadialReturn> rrr = MooseSharedNamespace::dynamic_pointer_cast<RecomputeRadialReturn>(getMaterialSharedPointerByName(models[i]));
+    StressUpdateBase * rrr = dynamic_cast<StressUpdateBase *>(&getMaterialByName(models[i]));
     if (rrr)
```

Comment:
That works, but there's no reason to turn a valid reference into a pointer. Just leave it as a reference intead:

`StressUpdateBase & rrr = dynamic_cast<StressUpdateBase &>(getMaterialByName(models[i]));`